src/utils/use-search.ts

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from "react";
+import { IFolder, IImage, metaDb, ROOT_FOLDER } from "./db";
+
+export const useSearch = () => {
```

Comment:
If possible, rename this function and the file to something more descriptive e.g. use-image-search